ross/defects/crack.py

```diff
@@ -0,0 +1,509 @@
+                             [0	        ,Koyz[3,0]	 ,0         ,Koyz[3,1]	,0 ,         0          ,0                  ,Koyz[3,2]          ,0      ,Koyz[3,3]          ,0          ,0],
+                             [Koxy[3,0]	,0           ,0         , 0	        ,Koxy[3,1]  ,0          ,Koxy[3,2]          ,0                  ,0      ,0                  ,Koxy[3,3]  ,0],
+                             [0	        ,0           ,0         , 0	        ,0	        ,0          ,0                  ,0                  ,0      ,0                  ,0	        ,0]])
+        # fmt: on
+        F_CRACK = np.zeros(self.ndof)
+
+        KK_CRACK = self.KK - KK_crack
+        FF_CRACK = (KK_CRACK).dot(self.positionsFis[self.dof_crack])
+        F_CRACK[self.dof_crack] = FF_CRACK
+
+        return FF_CRACK, F_CRACK
+
+    def _gasch(self, ap):
+        # Gasch
+        kme = (self.ko + self.kcx) / 2
+        kmn = (self.ko + self.kcz) / 2
+        kde = (self.ko - self.kcx) / 2
+        kdn = (self.ko - self.kcz) / 2
+
+        kee = kme + (4 / np.pi) * kde * (
+            np.cos(ap)
+            - np.cos(3 * ap) / 3
+            + np.cos(5 * ap) / 5
+            - np.cos(7 * ap) / 7
+            + np.cos(9 * ap) / 9
+            - np.cos(11 * ap) / 11
+            + np.cos(13 * ap) / 13
+            - np.cos(15 * ap) / 15
+            + np.cos(17 * ap) / 17
+            - np.cos(19 * ap) / 19
+            + np.cos(21 * ap) / 21
+            - np.cos(23 * ap) / 23
+            + np.cos(25 * ap) / 25
+            - np.cos(27 * ap) / 27
+            + np.cos(29 * ap) / 29
+            - np.cos(31 * ap) / 31
+            + np.cos(33 * ap) / 33
+            - np.cos(35 * ap) / 35
+        )
+
+        knn = kmn + (4 / np.pi) * kdn * (
+            np.cos(ap)
+            - np.cos(3 * ap) / 3
+            + np.cos(5 * ap) / 5
+            - np.cos(7 * ap) / 7
+            + np.cos(9 * ap) / 9
+            - np.cos(11 * ap) / 11
+            + np.cos(13 * ap) / 13
+            - np.cos(15 * ap) / 15
+            + np.cos(17 * ap) / 17
+            - np.cos(19 * ap) / 19
+            + np.cos(21 * ap) / 21
+            - np.cos(23 * ap) / 23
+            + np.cos(25 * ap) / 25
+            - np.cos(27 * ap) / 27
+            + np.cos(29 * ap) / 29
+            - np.cos(31 * ap) / 31
+            + np.cos(33 * ap) / 33
+            - np.cos(35 * ap) / 35
+        )
+
+        aux = np.array([[kee, 0], [0, knn]])
+
+        K = ((self.T_matrix.T).dot(aux)).dot(self.T_matrix)
+
+        return K
+
+    def _mayes(self, ap):
+        # Mayes
+
+        kee = 0.5 * (self.ko + self.kcx) + 0.5 * (self.ko - self.kcx) * np.cos(ap)
+
+        knn = 0.5 * (self.ko + self.kcz) + 0.5 * (self.ko - self.kcz) * np.cos(ap)
+
+        aux = np.array([[kee, 0], [0, knn]])
+
+        K = ((self.T_matrix.T).dot(aux)).dot(self.T_matrix)
+
+        return K
+
+    def _get_coefs(self, coef):
+
+        cwd = os.getcwd()
+        x = scipy.io.loadmat(cwd + "/tools/data/PAPADOPOULOS_c")
```

Comment:
In ROSS, the pattern is using `Path` library to manage paths (when saving or loading).
So, you could replace `os` and `sys` by `Path` and these lines by:
```python
cwd = Path(__file__).parent / "tools/data/PAPADOPOULOS_c"
x = scipy.io.loadmat(cwd)
```

Besides, using, `scipy.io.loadmat` intends to load a `MATLAB` file. I don't know if it's the best idea to work with. If it's possible, I'd recommend loading and .CSV file or something more "general".

Reply:
Thanks for the tips! And I will change the file (.mat) format. It is even more convenient.